### ISSUE TITLE
Multiple function calls asynchronously

### ISSUE THREAD

**Issue by avg0043** [reactphp/event-loop#135]
Hi ! I need to execute multiple calls to same function asynchronously. I have 2 projects (project1 and project2):

**Project 1:**

```
/**
* Loop an array of urls and call sub function.
**/
public function startFunction() {
  $finalResponse = [];
  $urls = ['www.google.es', 'www.github.com', 'www.bitbucket.org'];

  foreach ($urls as $url) {
    $res = $this->subFunction($url);    // subfunction call ( **IT MAY TAKE A LONG TIME !!** )
    $finalResponse[] = $res;
 }

 return $finalResponse;
}
```

```
/**
* Uses Factory loop to get the Promise returned by finalRequest function.
**/
private function subFunction($url) {
  $loop = \React\EventLoop\Factory::create();
  $classA = new Project2\ClassA();
  $finalResponse = null;

  // project 2 function call
  $classA->finalRequest($url)->then(function($response) use(   
     &$finalResponse
  ) {
     $finalResponse = $response;
  })
  
  return $finalResponse;
}
```

**Project 2:**

```
/**
* Makes an React\HttpClient request (GET) to sent url and return his value inside a Promise.
**/
public function finalRequest($url) {
   $generalDeferred = new Deferred();
   $generalPromise = $generalDeferred->promise();

   // make React\HttpClient request
   $request = $client->request('GET', $url);
   $request->on('response', function ($response) use($generalDeferred) {
   $response->on('data', function ($response) {
        $generalDeferred->resolve($response);
      });
   });
  $request->end();

  return $generalPromise;
}
```

Problem is that on every subFunction($url) call, the program stops until the sub Function gets the response, but I need to do this asynchronously because this subFunction could take many seconds.
So I would like to launch all subFunction($url) calls at the same time, and get all responses asynchronously.

It's possible solve this problem? Thanks.

**Comment by kelunik** [comment]
There are two options:
 - Refactor `subFunction` to use non-blocking I/O.
 - Use child processes to execute the `subFunction` in parallel.

**Comment by avg0043** [comment]
I have read about two options but I doesn't know how can I adapt it to my code. 
About [Child processes](https://github.com/reactphp/child-process) I have only found examples that uses command lines commands.. so how can use it for my subFunction ?
Can you show me an example code or reference link? 
Thanks.

**Comment by clue** [comment]
It looks like this may have been answered in your stackoverflow question already: https://stackoverflow.com/questions/47769258/multiple-function-calls-asynchronously-in-php

I'll assume this is resolved and will close this for now, please feel free to report back otherwise :+1:
